xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -422,14 +421,14 @@ bool CAESinkAUDIOTRACK::Initialize(AEAudioFormat &format, std::string &device)
       switch (m_format.m_streamInfo.m_type)
       {
         case CAEStreamInfo::STREAM_TYPE_TRUEHD:
-          m_min_buffer_size = MAX_RAW_AUDIO_BUFFER_HD;
-          m_format.m_frames = m_min_buffer_size;
-          rawlength_in_seconds = 8 * m_format.m_streamInfo.GetDuration() / 1000; // on average
+          m_min_buffer_size = MAX_RAW_AUDIO_BUFFER_TRUEHD;
+          m_format.m_frames = m_min_buffer_size / 2;
+          rawlength_in_seconds = 0.04;
```

Comment:
To avoid magic numbers seems better replace 0.04 with:

`2 * m_format.m_streamInfo.GetDuration() / 1000;`

Also for me seems more clear simply define only 61440 number. I mean:

```c++
const unsigned int TRUEHD_SIZE = 61440;

m_min_buffer_size = 2 * TRUEHD_SIZE;
m_format.m_frames = TRUEHD_SIZE;
rawlength_in_seconds = 2 * m_format.m_streamInfo.GetDuration() / 1000;
```

With this is clear that is used a buffer two times TrueHD size.

Reply:
I continue on at2025final2 - will ping you again when I am happy - then we can take over the bits we want in here. Thanks again.

Reply:
Okay, I'll continue tomorrow.